ingest/src/test/java/com/connexta/ingest/IngestApplicationIntegrationTest.java

```diff
@@ -49,9 +55,59 @@ public void beforeEach() {
     mvc = MockMvcBuilders.webAppContextSetup(wac).build();
   }
 
+  @After
+  public void afterEach() {
+    reset(restTemplate);
+    reset(mockS3Client);
+  }
+
   @Test
   public void testContextLoads() {}
 
+  @Test
+  public void testS3Unavailable() throws Exception {
+    // given
+    when(mockS3Client.putObject(any(PutObjectRequest.class), any(RequestBody.class)))
+        .thenThrow(SdkServiceException.builder().build());
+
+    mvc.perform(
+            multipart("/ingest")
+                .file("file", "some-content".getBytes())
+                .param("fileSize", "10")
+                .param("fileName", "file")
+                .param("title", "qualityTitle")
+                .param("mimeType", "plain/text")
+                .header("Accept-Version", "1.2.1")
+                .accept(MediaType.APPLICATION_JSON)
+                .contentType(MediaType.MULTIPART_FORM_DATA))
+        .andExpect(status().is5xxServerError());
+
+    // verify
+    verifyZeroInteractions(restTemplate);
+  }
+
+  @Test
+  public void testS3UnableToStore() throws Exception{
```

Comment:
https://github.com/connexta/multi-int-store/pull/64#discussion_r296045332 I think this is right. We should have a test method for every exception that can be thrown. This is easier with parameterized tests, of course.